core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -188,7 +193,7 @@ public void initialize(String name, Map<String, String> unresolved) {
               client, tokenRefreshExecutor(), token, expiresAtMillis(mergedProps), catalogAuth);
     }
 
-    this.io = newFileIO(mergedProps);
+    this.io = newFileIO(mergedProps, null);
```

Comment:
I'd probably make this an empty session context.

Reply:
Done